### PR TITLE
Upgrading laravel integration

### PR DESCRIPTION
**Pull request by shea786**

### CODE REVIEW

File: api/app/Modules/Accounts/Http/Controllers/LibraryController.php [hunk 1]
```diff
@@ -27,11 +27,9 @@ public function tracks(Request $request): JsonResponse
         return $this->respondWithPaginator($tracks, new TrackTransformer());
     }
 
-    public function getTrackIds()
+    public function getTrackIds(): JsonResponse
     {
-        $tracks = $this->getUser()->savedTracks()->get(['saveable_id'])->map(function ($result) {
-            return $result->saveable_id;
-        })->unique()->values();
+        $tracks = $this->getUser()->savedTracks()->get(['saveable_id'])->unique()->values();
```
Comment (szainmehdi) [inline api/app/Modules/Accounts/Http/Controllers/LibraryController.php:32]: Does this change the response?

**Review by szainmehdi — COMMENTED** [review]
Do you need help getting CI checks to pass?

And have you verified that the FE doesn't require any changes for this PR?

File: api/app/Modules/Accounts/Http/Controllers/LibraryController.php (hunk 1 above)
Comment (shea786) [reply api/app/Modules/Accounts/Http/Controllers/LibraryController.php:32]: @szainmehdi  This method is not being used so maybe we can remove it ?

Previously it was sending only the saveable_id but now it is sending saveable_id as well as the saveable object

Psalm kept complaining about how saveable_id is not a property on the class, as we technically don't have a saveable model.
But again, I think we can remove this method completely as we are not using it at all, I have checked the frontend code too and no use of this api method `/v1/tracks/ids`

**Comment by shea786** [comment]
> Do you need help getting CI checks to pass?
> 
> And have you verified that the FE doesn't require any changes for this PR?

@szainmehdi  the only part which is failing currently is the psalm errors I have informed you about.
I have verified the frontend on my local machine and all looks good.
Waiting for this PR build to get passed so I can test in integration too

File: api/app/Modules/Accounts/Http/Controllers/LibraryController.php (hunk 1 above)
Comment (szainmehdi) [reply api/app/Modules/Accounts/Http/Controllers/LibraryController.php:32]: We're definitely using it, though it's a little hard to follow the code right now to find where. But, the endpoint definition in the frontend is here
https://github.com/nawhas/nawhas/blob/d04cc2c1b276a1f39a9fdf4cf77ca66581aa36c7/nuxt/api/library.ts#L47-L49

and this method is used here
https://github.com/nawhas/nawhas/blob/d04cc2c1b276a1f39a9fdf4cf77ca66581aa36c7/nuxt/store/library.ts#L34-L53

and that is called in these places
https://github.com/nawhas/nawhas/blob/d04cc2c1b276a1f39a9fdf4cf77ca66581aa36c7/nuxt/middleware/initialize.ts#L11
and https://github.com/search?q=repo%3Anawhas%2Fnawhas%20library%2FgetTrackIds&type=code

Comment (szainmehdi) [inline api/app/Modules/Accounts/Http/Controllers/LibraryController.php:32]: We need to revert this change and address whatever linting error you were facing

**Review by szainmehdi — CHANGES_REQUESTED** [review]
Since the

File: api/app/Modules/Accounts/Http/Controllers/LibraryController.php (hunk 1 above)
Comment (shea786) [reply api/app/Modules/Accounts/Http/Controllers/LibraryController.php]: Change reverted

File: api/app/Modules/Core/Transformers/Transformer.php [hunk 2]
```diff
@@ -41,10 +41,10 @@ protected function timestamps(TimestampedEntity $entity): MergeValue
...
 
     /**
-     * @return \League\Fractal\Resource\NullResource|Primitive
+     * @return NullResource
      */
-    protected function null()
+    protected function null(): NullResource
     {
-        return $this->primitive(null);
+        return new NullResource();
```
Comment (szainmehdi) [inline api/app/Modules/Core/Transformers/Transformer.php]: This is causing issues in the frontend. For tracks with no lyrics, the response has changed from:

```
"lyrics": null,
```

to 
```
"lyrics": [],
```

That's why this URL is breaking, along with many others: https://pr-592.intg.k8s.nawhas.com/reciters/farhan-ali-waris/albums/2008/tracks/ab-tera-bichona

I've pushed a commit (234f8563) to fix this, but just an FYI

**Review by szainmehdi — APPROVED** [review]
Looks good now!
